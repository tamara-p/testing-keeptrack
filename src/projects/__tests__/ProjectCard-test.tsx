import { render, screen } from '@testing-library/react';
import ProjectCard from '../ProjectCard';
import { Project} from '../Project';
import React from 'react';
import { MemoryRouter} from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

describe("<ProjectCard/>", () =>{
    let project: Project;
    let handleEdit: jest.Mock;

    const setup = () => 
        render(
            <MemoryRouter>
        <ProjectCard project = {project} onEdit={handleEdit} />
        </MemoryRouter>
        );

    beforeEach(() =>{
        project = new Project({
            id: 1,
            name: "Random name",
            description: "describing the thing",
            budget: 100,
        });
        handleEdit = jest.fn();
    });

    it("should initially render", () => {
        setup();
    });

    it('renders project properly', () => {
       
            setup();
        
        expect(screen.getByRole('heading')).toHaveTextContent(project.name);
        //screen.debug(document);
        screen.getByText(/describing the thing\.\.\./i);
        screen.getByText(/budget : 100/i);
        // searching for a text in the projcard
        
    });

    it('handler called when edit clicked', async () =>{
        
         setup();
        
        const view = userEvent.setup();
        await view.click(screen.getByRole('button', { name: /edit/i}));
        expect(handleEdit).toBeCalledTimes(1);
        expect(handleEdit).toBeCalledWith(project);
    });

    test('snapshot', () => {
        const tree = renderer
            .create(
                <MemoryRouter>
                <ProjectCard project = {project} onEdit={handleEdit} />
                </MemoryRouter>
            )
            .toJSON();
            expect(tree).toMatchSnapshot();
    }
    );
});