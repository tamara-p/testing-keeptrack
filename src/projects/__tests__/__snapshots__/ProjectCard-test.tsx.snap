// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCard/> snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="Random name"
    src=""
  />
  <section
    className="section dark"
  >
    <a
      href="/projects/1"
      onClick={[Function]}
    >
      <h5
        className="strong"
      >
        <strong>
           
          Random name
        </strong>
      </h5>
      <p>
         
        describing the thing...
      </p>
      <p>
         Budget : 
        100
      </p>
    </a>
    <button
      className="bordered"
      onClick={[Function]}
    >
      <span
        className="icon-edit"
      />
      Edit
    </button>
  </section>
</div>
`;
